server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/StarTreeUtils.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.index.compositeindex.datacube.startree.utils;
+
+import org.apache.lucene.index.DocValuesType;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.VectorEncoding;
+import org.apache.lucene.index.VectorSimilarityFunction;
+import org.opensearch.index.compositeindex.datacube.startree.aggregators.MetricAggregatorInfo;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Util class for building star tree
+ *
+ * @opensearch.experimental
+ */
+public class StarTreeUtils {
+
+    private StarTreeUtils() {}
+
+    public static final int ALL = -1;
+
+    /**
+     * The suffix appended to dimension field names in the Star Tree index.
+     */
+    public static final String DIMENSION_SUFFIX = "dim";
+
+    /**
+     * The suffix appended to metric field names in the Star Tree index.
+     */
+    public static final String METRIC_SUFFIX = "metric";
+
+    /**
+     * Returns the full field name for a dimension in the star-tree index.
+     *
+     * @param starTreeFieldName star-tree field name
+     * @param dimensionName     name of the dimension
+     * @return full field name for the dimension in the star-tree index
+     */
+    public static String fullyQualifiedFieldNameForStarTreeDimensionsDocValues(String starTreeFieldName, String dimensionName) {
+        return starTreeFieldName + "_" + dimensionName + "_" + DIMENSION_SUFFIX;
```

Comment:
Is there a use case where fully qualified name needs to be converted back to the tokens?

Reply:
I don't see a usecase like that coming. We rely on the original field names to construct unique doc value names, not vice versa